cli/src/cli_util.rs

```diff
@@ -2248,12 +2248,13 @@ impl DiffSelector {
         left_tree: &MergedTree,
         right_tree: &MergedTree,
         matcher: &dyn Matcher,
-        instructions: Option<&str>,
+        format_instructions: impl FnOnce() -> String,
     ) -> Result<MergedTreeId, CommandError> {
         match self {
             DiffSelector::NonInteractive => Ok(restore_tree(right_tree, left_tree, matcher)?),
             DiffSelector::Interactive(editor) => {
-                Ok(editor.edit(left_tree, right_tree, matcher, instructions)?)
+                let instructions = format_instructions();
+                Ok(editor.edit(left_tree, right_tree, matcher, Some(&instructions))?)
```

Comment:
nit: `format_instructions` could be passed in to `.edit()`. Not all editors use instructions.

BTW, instruction templates might have to be defined per editor, but I don't know how many customization points (and complexities) we'll want.

Reply:
That's a good idea. I updated it, could you take a look and see if it looks good when you get a chance?